spec.md

```diff
@@ -173,6 +173,86 @@ minimally trusted storage servers is required.
 
 ## Authorization
 
+The ability to do cross-domain, operator-independent, standardized cloud storage
+operations requires an authorization system that is:
+
+* Modular and layered (for future agility / upgradability)
+* Not limited to traditional domain-based "usernames and passwords"
+* A hybrid, using object capability principles at its baseline, but also able
+  to provide ACL or RBAC-like functionality for user convenience. In other words,
+  the system needs to support both "anyone with the link can..." and "these are
+  specific people and groups allowed to..." styles of access control
+* Compatible with cross-domain replication
+* Compatible with end-to-end client side encryption (but also not rely on
+  encryption as the sole authorization method)
+* "Private by default".
+  That is, by default, unless otherwise specified, only the controller of a
+  space (or of a collection or resource) is authorized to perform any operation
+  (read, write, delete, etc)
+
+As the state of the art in cross-domain authorization advances, we expect there
+to be multiple profiles and specs that could be used to perform WAS API
+calls. However, to start with, this specification will focus on a single minimal
+authorization profile.
+
+### WAS Authorization Profile v0.1
+
+The initial W.A.S. Authorization Profile uses the following specifications.
+
+1. Identity (for controllers or clients/agents): [DID 1.1](https://www.w3.org/TR/did-1.1/)
+   or [CID 1.0](https://www.w3.org/TR/cid-1.0/) identifiers.
+2. Capability data model: [Authorization Capabilities for Linked Data v0.3](https://w3c-ccg.github.io/zcap-spec/)
+3. Protocol for obtaining authorization: Out of scope (implementers are encouraged
+   to use VC-API, OpenId4VP, or GNAP, as appropriate)
+4. Proof of Possession / authorization invocation: [HTTP Signatures (Cavage
+   draft 12)](https://datatracker.ietf.org/doc/html/draft-cavage-http-signatures)
+5. Access Control / Policy language data model: TBD
+
+#### Space `controller` and the Root of Trust
+
+Space `controller`s MUST be in the form of a [DID](https://www.w3.org/TR/did-1.1/)
+or a [CID](https://www.w3.org/TR/cid-1.0/) identifier.
+
+For minimal compatibility, all WAS implementations MUST support the
+[`did:key` DID Method](https://w3c-ccg.github.io/did-key-spec/), using the
+Multikey encoding of `Ed25519` elliptic curve keys, as specified in the
+[Multikey section of the CID spec](https://www.w3.org/TR/cid-1.0/#Multikey)
+as the space `controller`.
+
+When a space is created via an HTTP [POST](#http-api-post-spaces) or
+[PUT](#http-api-put-space-space_id) operation, the controller for that space
+is set, either implicitly or explicitly. 
+
+Implicitly, if no `controller` is specified in the PUT or POST create space
```

Comment:
... but what if there is no controller ID in context? so far i've been supporting `curl -X POST $WAS/spaces/` to create a space with no controller. This requirement would require me to determine the controller for that request.... ok but there is no controller in that request.

Reply:
I'm not sure a space with no controller makes sense?

Reply:
Or, to put it a different way, spaces with no controller -- out of scope for this spec. (not comformant)

Reply:
That's fine. From a spec readability perspective, it still doesn't really answer: This text requires the spec to determine the controller, but doesn't say *how*. And it's a common case that there may be no controller in context.... so it could be helpful to give the server guidance on what to do in that case, or how to avoid that case. e.g. "If no controller can be determined, the server must not fulfill the request to create a space" or "To ensure this requirement is satisfied even when the request does not indicate a controller, the server may be configured with a default controller to use when no other controller can be determined".

Reply:
For now, throw a 400 if there is no controller